### PR TITLE
Task completed: GP-81 Death Menu

### PR DESCRIPTION
**Pull request by Kjusok**
Added DeathMenu prefab

On the screen when opening the Death Menu:
* Game Over Header
* Done button
* The time it took the player to level up
* After pressing the Done button, the MainMenu is loaded

### CODE REVIEW

File: Assets/Scripts/UI/Views/Popups/DeathMenu/DeathMenu.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+            
+            if (hours > 0)
+            {
+                text.text = string.Format("{0:D2}", hours) + text.text;
+            }
+        }
+
+        public void PressDoneButton()
+        {
+            _gameRun.Finish();
```
Comment (ega-ega) [inline Assets/Scripts/UI/Views/Popups/DeathMenu/DeathMenu.cs:53]: Add invoke Close()

File: Assets/Scripts/UI/Views/Popups/DeathMenu/DeathMenu.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        {
+            _pauseManager = pause;
+            _gameRun = gameRun;
+            _gameTime = gameTime;
+        }
+
+        
+        private void Start()
+        {
+            _pauseManager.Paused = true;
```
Comment (ega-ega) [inline Assets/Scripts/UI/Views/Popups/DeathMenu/DeathMenu.cs:30]: Pls add set to false after this popup closed

File: Assets/Scripts/UI/Views/Popups/DeathMenu/DeathMenuObserver.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+        
+        private void OnEnable()
+        {
+            _player.Dead += ToggleDeathMenu;
+        }
+        
+        private void OnDisable()
+        {
+            _player.Dead += ToggleDeathMenu;
```
Comment (ega-ega) [inline Assets/Scripts/UI/Views/Popups/DeathMenu/DeathMenuObserver.cs]: its mistake. Redo to unsubscribe "-="

**Review by ega-ega — CHANGES_REQUESTED** [review]

File: Assets/Scripts/UI/Views/Popups/DeathMenu/DeathMenu.cs (hunk 1 above)
Comment (Kjusok) [reply Assets/Scripts/UI/Views/Popups/DeathMenu/DeathMenu.cs:53]: Done

File: Assets/Scripts/UI/Views/Popups/DeathMenu/DeathMenu.cs (hunk 2 above)
Comment (Kjusok) [reply Assets/Scripts/UI/Views/Popups/DeathMenu/DeathMenu.cs:30]: Done

File: Assets/Scripts/UI/Views/Popups/DeathMenu/DeathMenuObserver.cs (hunk 3 above)
Comment (Kjusok) [reply Assets/Scripts/UI/Views/Popups/DeathMenu/DeathMenuObserver.cs]: Done

File: Assets/Scripts/Gameplay/IPlayer.cs [hunk 4]
```diff
@@ -1,9 +1,11 @@
 ﻿using UnityEngine;
-
+using System;
 namespace Gameplay
 {
     public interface IPlayer
     {
+        public event Action Dead;
```
Comment (ega-ega) [inline Assets/Scripts/Gameplay/IPlayer.cs]: Pls remove public word

**Review by ega-ega — CHANGES_REQUESTED** [review]

File: Assets/Scripts/Gameplay/IPlayer.cs (hunk 4 above)
Comment (Kjusok) [reply Assets/Scripts/Gameplay/IPlayer.cs]: Done

**Review by frnzee — APPROVED** [review]

**Review by ega-ega — APPROVED** [review]

**Pull request merged**